### ISSUE TITLE
search_fields only matches root instance with HierarchicalModelAdmin

### ISSUE THREAD

**Issue by elver** [idlesign/django-admirarchy#8]
I've got hierarchical categories like this:

```
@admin.register(Category)
class CategoryAdmin(HierarchicalModelAdmin, SummernoteModelAdmin):
    hierarchy = AdjacencyList("parent_category")
    summernote_fields = ("content",)
    search_fields = ["full_name", "short_name",]
```

And while a search box is displayed, searching only matches fields on the topmost category object. How do I fix this so that the search box also matches subcategories?

**Comment by idlesign** [comment]
This search behaviour is known. 
It could be reimplemented somehow, if we define rules for search.

Now it's not clear whether we want to search 
a. in a current node or 
b. in all nodes or 
c. in subnodes.

**Comment by elver** [comment]
I want to search in all nodes and I am happy with a flat list as a result. What can I do?

**Comment by idlesign** [comment]
Nothing could be done so far. Design decision and implementation are required, as I've already said.
Can't promise any of these in short term, but I'll keep the issue in mind.

**Comment by idlesign** [comment]
This seems to be resolved in 1.2.0.

The logic is now as follows:
* When in root search works as global: all relevant items are shown despite of parents.
* When in non-root: only immediate ancestors are searched.

Considered closed. 
Feel free to reopen if required.
